### PR TITLE
Replace Makefile and MS-Windows scripts with bb tasks

### PR DESCRIPTION
**Pull request by ikappaki**
Hi,

could you please consider patch to replace the Makefile and windows scripts with bb tasks. This is first step of the attempt discussed in #1211 to align the MS-Windows builds and dev cycle with that of Linux.

The `make` tool is not readily available nor fully compatible with the MS-Windows, thus windows users need to do extra work to develop clojure-lsp, thus it is much easier if we have a common build tool.

Changes include
- Replicating the `Makefile` targets as close as possible in `scripts/make.clj`.
- Changing the GH actions and other script to replace `make xxx` invocations with `bb xxx` invocations.
- Changing documentation to refer to `bb xxx` instead of `make xxx`
- Removing `Makefile` and the `native-windows-compile.bat` script.

The user experience should remain the same as before, but instead of running `make` users should now run `bb`.  

The only thing I could not replicate is the plain `make` command which invokes `make debug-cli`. Running `bb` will just lists the available tasks instead. Should we add a `bb cli` alias instead for `bb debug-cli` to easy the transition?

The following workflows have been tested to work:
- `ci.yml`.
- `release.yml`, except the release jobs because they require tokens.
- `nightly.hml`, except those parts that require GH tokens.

The following have not been tested:
- `cirrus.yml`  - not sure how to test this.
- `bump_downstreams.yml`, because it requires authentication.

All the untested elements mentioned above shall be tested I believe on commit by the build infrastructure. Any unexpected issues should be easy to fix, since the changes are very simple in nature (installing the bb script where it does not exist, changing `make` to `bb`).

Thanks

- [x] I created an issue to discuss the problem I am trying to solve or an open issue already exists.
- [x] I added a new entry to [CHANGELOG.md](https://github.com/clojure-lsp/clojure-lsp/blob/master/CHANGELOG.md)
- [x] I updated documentation if applicable (`docs` folder)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@
   - Add support to decompile jar as a java project when finding the definition of a java class, allowing to have LSP features on that java project. #1187
   - Add refactorings similar to `Sort map`: `Sort vector`, `Sort list`, `Sort set`, and `Sort clauses` for functions like `assoc`. #1155
 
+- Build
+  - Replace Makefile and MS-Windows scripts with bb tasks. #1211
```
Comment (ericdallo) [inline CHANGELOG.md:14]: Usually, we add an entry to refer to changes that affect users, this is not one that will affect the final user so not sure it's worth polluting the changelog with these changes, WDYT?

**Review by ericdallo — APPROVED** [review]
I think we could remove the changelog, besides that looks really good, thank you for the help!

File: CHANGELOG.md (hunk 1 above)
Comment (ikappaki) [reply CHANGELOG.md]: Done, hopefully existing developers won't have much trouble figuring out the move to bb :)

**Comment by ikappaki** [comment]
> I think we could remove the changelog, besides that looks really good, thank you for the help!

Done! thank you for the review!

**Pull request merged**